File: apprefurbished/src/main/java/com/androidcollider/easyfin/transactions/list/TransactionsPresenter.java
package com.androidcollider.easyfin.transactions.list;

import androidx.annotation.Nullable;

/**
 * @author Ihor Bilous
 */

public class TransactionsPresenter implements TransactionsMVP.Presenter {

    @Nullable
    private TransactionsMVP.View view;
    private TransactionsMVP.Model model;


    public TransactionsPresenter(TransactionsMVP.Model model) {
        this.model = model;
    }

    @Override
    public void setView(@Nullable TransactionsMVP.View view) {
        this.view = view;
    }

    @Override
    public void loadData() {
        model.getTransactionAndTransactionCategoriesLists()
                .subscribe(
                        pair -> {
                            if (view != null) {
                                view.setTransactionAndTransactionCategoriesLists(pair.first, pair.second.first, pair.second.second);
                            }
                        },
                        Throwable::printStackTrace
                );
    }

    @Override
    public void getTransactionById(int id) {
        model.getTransactionById(id)
                .subscribe(
                        transaction -> {
                            if (view != null) {
                                view.goToEditTransaction(transaction);
                            }
                        },
                        Throwable::printStackTrace
                );
    }

    @Override
    public void deleteTransactionById(int id) {
        model.deleteTransactionById(id)
                .subscribe(
                        aBoolean -> {
                            if (aBoolean && view != null) {
                                view.deleteTransaction();
                            }
                        },
                        Throwable::printStackTrace
                );
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/transactions/list/TransactionsPresenter.kt
package com.androidcollider.easyfin.transactions.list

import com.androidcollider.easyfin.common.models.Transaction
import com.androidcollider.easyfin.common.models.TransactionCategory

/**
 * @author Ihor Bilous
 */
class TransactionsPresenter(private val model: TransactionsMVP.Model) : TransactionsMVP.Presenter {

    private var view: TransactionsMVP.View? = null
    override fun setView(view: TransactionsMVP.View?) {
        this.view = view
    }

    override fun loadData() {
        model.transactionAndTransactionCategoriesLists
            .subscribe({ pair: Pair<List<TransactionViewModel>,
                    Pair<List<TransactionCategory>,
                            List<TransactionCategory>>> ->
                view?.setTransactionAndTransactionCategoriesLists(
                    pair.first,
                    pair.second.first,
                    pair.second.second
                )
            },
                { obj: Throwable -> obj.printStackTrace() })
    }

    override fun getTransactionById(id: Int) {
        model.getTransactionById(id)
            .subscribe(
                { transaction: Transaction ->
                    view?.goToEditTransaction(transaction)
                })
            { obj: Throwable -> obj.printStackTrace() }
    }

    override fun deleteTransactionById(id: Int) {
        model.deleteTransactionById(id)
            .subscribe(
                { aBoolean: Boolean ->
                    if (aBoolean) {
                        view?.deleteTransaction()
                    }
                })
            { obj: Throwable -> obj.printStackTrace() }
    }
}